arbiter-core/src/environment.rs

```diff
@@ -32,63 +50,84 @@ pub enum State {
     Stopped,
 }
 
+/// The environment struct.
 pub struct Environment {
+    /// label for the environment
     pub label: String,
     pub(crate) state: State,
     pub(crate) evm: EVM<CacheDB<EmptyDB>>,
+    /// Connection to the environment
     pub connection: Connection,
     /// Clients (Agents) in the environment
-    pub clients: Vec<Arc<Agent<RevmMiddleware>>>,
-    // pub deployed_contracts: HashMap<String, Contract<RevmMiddleware>>,
+    pub clients: HashMap<String, Agent<RevmMiddleware>>,
+    /// expected events per block
+    pub lambda: f64,
 }
 
+/// Connection struct for the [`Environment`].
 #[derive(Debug, Clone)]
 pub struct Connection {
     pub(crate) tx_sender: TxEnvSender,
     tx_receiver: TxEnvReceiver,
     pub(crate) event_broadcaster: Arc<Mutex<EventBroadcaster>>,
+    /// expected events per block
+    pub tx_per_block: Arc<AtomicUsize>,
 }
 
 impl Environment {
+    /// Create a new [`Environment`].
     pub(crate) fn new(label: String) -> Self {
         let mut evm = EVM::new();
         let db = CacheDB::new(EmptyDB {});
         evm.database(db);
         evm.env.cfg.limit_contract_code_size = Some(0x100000); // This is a large contract size limit, beware!
         evm.env.block.gas_limit = U256::MAX;
-        let transaction_channel = unbounded::<(ToTransact, TxEnv, Sender<ExecutionResult>)>();
+        let transaction_channel = unbounded::<(ToTransact, TxEnv, Sender<RevmResult>)>();
         let connection = Connection {
             tx_sender: transaction_channel.0,
             tx_receiver: transaction_channel.1,
             event_broadcaster: Arc::new(Mutex::new(EventBroadcaster::new())),
+            tx_per_block: Arc::new(AtomicUsize::new(0)),
         };
         Self {
             label,
             state: State::Stopped,
             evm,
             connection,
-            clients: vec![],
+            clients: HashMap::new(),
+            lambda: 0.0,
         }
     }
-    // TODO: We need to make this the way to add agents to the environment.
-    // in `agent.rs` we have `new_simulation_agent` which should probably just be called from this function instead.
-    // OR agents can be created (without a connection?) and then added to the environment where they will gain a connection?
-    pub fn add_agent(&mut self, agent: Agent<RevmMiddleware>) {
-        self.clients.push(Arc::new(agent));
+    /// Set the expected events per block
+    pub(crate) fn configure_lambda(&mut self, lambda: f64) {
```

Comment:
Why do this when we can just set lambda in the constructor?

Reply:
Unless we want to be able to change it as we go through a sim? Either way, we should probably set a starting `lambda` instead of just having it be 0. 

We can also make the `lambda = 0` case one where we don't even update blocks.

Reply:
So I made it an option and only applied it if it is some. I kept the ability for users to configure it if they want which i think is good for now.